src/pubsub/ua_pubsub_ns0.c

```diff
@@ -248,16 +266,22 @@ addPubSubConnectionConfig(UA_Server *server, UA_PubSubConnectionDataType *pubsub
     //connectionConfig.enabled = pubSubConnectionDataType.enabled;
     UA_Variant_setScalar(&connectionConfig.address, &networkAddressUrlDataType,
                          &UA_TYPES[UA_TYPES_NETWORKADDRESSURLDATATYPE]);
-    if(pubsubConnectionDataType->publisherId.type == &UA_TYPES[UA_TYPES_UINT32]){
-        connectionConfig.publisherId.numeric = * ((UA_UInt32 *) pubsubConnectionDataType->publisherId.data);
-    } else if(pubsubConnectionDataType->publisherId.type == &UA_TYPES[UA_TYPES_STRING]){
-        connectionConfig.publisherIdType = UA_PUBSUB_PUBLISHERID_STRING;
-        UA_String_copy((UA_String *) pubsubConnectionDataType->publisherId.data, &connectionConfig.publisherId.string);
+
+    if (pubsubConnectionDataType->publisherId.type == &UA_TYPES[UA_TYPES_BYTE]) {
+        connectionConfig.publisherIdType = UA_PUBLISHERIDTYPE_BYTE;
+    } else if (pubsubConnectionDataType->publisherId.type == &UA_TYPES[UA_TYPES_UINT16]) {
+        connectionConfig.publisherIdType = UA_PUBLISHERIDTYPE_UINT16;
+    } else if (pubsubConnectionDataType->publisherId.type == &UA_TYPES[UA_TYPES_UINT32]) {
+        connectionConfig.publisherIdType = UA_PUBLISHERIDTYPE_UINT32;
+    } else if (pubsubConnectionDataType->publisherId.type == &UA_TYPES[UA_TYPES_UINT64]) {
+        connectionConfig.publisherIdType = UA_PUBLISHERIDTYPE_UINT64;
+    } else if (pubsubConnectionDataType->publisherId.type == &UA_TYPES[UA_TYPES_STRING]) {
+        connectionConfig.publisherIdType = UA_PUBLISHERIDTYPE_STRING;
     } else {
-        UA_LOG_WARNING(&server->config.logger, UA_LOGCATEGORY_SERVER, "Unsupported PublisherId Type used.");
-        //TODO what's the best default behaviour here?
-        connectionConfig.publisherId.numeric = 0;
+        UA_LOG_ERROR(&server->config.logger, UA_LOGCATEGORY_SERVER, "Unsupported PublisherId Type used.");
+        return UA_STATUSCODE_BADCONFIGURATIONERROR;
     }
+    memcpy(&connectionConfig.publisherId, pubsubConnectionDataType->publisherId.data, sizeof(connectionConfig.publisherId));
```

Comment:
This implies a shallow copy of the string id.
That is intended?

Reply:
yes shallow copy is intended, because deep copy happens at next line within `UA_Server_addPubSubConnection`

`removeConnectionAction` calls `UA_Server_removePubSubConnection()` calls `UA_PubSubConnection_clear()` calls `UA_PubSubConnectionConfig_clear()` which frees string